packages/ui/test/forum/modals/DeleteThreadModal.test.tsx

```diff
@@ -0,0 +1,143 @@
+import { DeleteThreadModal, DeleteThreadModalCall } from '@/forum/modals/DeleteThreadModal'
+import { MembershipContext } from '@/memberships/providers/membership/context'
+import { MyMemberships } from '@/memberships/providers/membership/provider'
+import { seedMember } from '@/mocks/data'
+import rawMembers from '@/mocks/data/raw/members.json'
+import { seedForumCategory, seedForumThread } from '@/mocks/data/seedForum'
+import { randomBlock } from '@/mocks/helpers/randomBlock'
+
+import { getButton } from '../../_helpers/getButton'
+import { mockCategories, mockThreads } from '../../_mocks/forum'
+import { alice, bob } from '../../_mocks/keyring'
+import { getMember } from '../../_mocks/members'
+import { MockKeyringProvider, MockQueryNodeProviders } from '../../_mocks/providers'
+import { setupMockServer } from '../../_mocks/server'
+import {
+  stubApi,
+  stubDefaultBalances,
+  stubTransaction,
+  stubTransactionFailure,
+  stubTransactionSuccess,
+} from '../../_mocks/transactions'
+
+jest.mock('@/common/hooks/useQueryNodeTransactionStatus', () => ({
+  useQueryNodeTransactionStatus: () => 'confirmed',
+}))
+
+describe('UI: DeleteThreadModal', () => {
+  const api = stubApi()
+  const txPath = 'api.tx.forum.deleteThread'
+  const modalData: ModalCallData<DeleteThreadModalCall> = {
+    thread: {
+      id: '1',
+      title: 'Example Thread',
+      categoryId: '1',
+      authorId: '0',
+      isSticky: false,
+      createdInBlock: randomBlock(),
+      tags: [],
+      visiblePostsCount: 5,
+      status: { __typename: 'ThreadStatusActive' },
+    },
+  }
+
+  const useModal: UseModal<any> = {
+    hideModal: jest.fn(),
+    showModal: jest.fn(),
+    modal: null,
+    modalData,
+  }
+  const useMyMemberships: MyMemberships = {
+    active: undefined,
+    members: [],
+    setActive: (member) => (useMyMemberships.active = member),
+    isLoading: false,
+    hasMembers: true,
+    helpers: {
+      getMemberIdByBoundAccountAddress: () => undefined,
+    },
+  }
+  let useAccounts: UseAccounts
+
+  const server = setupMockServer({ noCleanupAfterEach: true })
+
+  beforeAll(async () => {
+    await cryptoWaitReady()
+    rawMembers.slice(0, 2).map((member) => seedMember(member, server.server))
+    seedForumCategory(mockCategories[0], server.server)
+    seedForumThread(mockThreads[0], server.server)
+    useMyMemberships.members = [getMember('alice'), getMember('bob')]
+    useMyMemberships.setActive(getMember('alice'))
+    useAccounts = {
+      isLoading: false,
+      hasAccounts: true,
+      allAccounts: [alice, bob],
+    }
+  })
+
+  beforeEach(async () => {
+    stubDefaultBalances(api)
+  })
+
+  it('Requirements passed', async () => {
+    stubTransaction(api, txPath, 10)
+    renderModal()
+    expect(screen.findByText('modals.deleteThread.description')).toBeDefined()
+    expect(screen.findByText('modals.deleteThread.buttonLabel')).toBeDefined()
+  })
```

Comment:
Lately there has been several issues with modals not passing the right parameters to transactions (which typescript doesn't check). From now on it would be good to test this in every new transaction.

Here I think the best would be to add something like:
```ts
const txMock = api.api.tx.forum.deleteThread as unknown as jest.Mock
...
const [userId, categoryId, threadId, hide] = last(txMock.mock.calls)
expect(userId).toBe(...
```
to this `Requirements passed` test. A more complete example here: [packages/ui/test/bounty/modals/SubmitJudgementModal.test.tsx](https://github.com/Joystream/pioneer/blob/8e50fe7cf87a220a6e76203183530c47bd1e9ad3/packages/ui/test/bounty/modals/SubmitJudgementModal.test.tsx#L127)